sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -138,6 +142,8 @@ def _validate_params(self, for_partial_fit=False):
             raise ValueError("max_iter must be > zero. Got %f" % self.max_iter)
         if not (0.0 <= self.l1_ratio <= 1.0):
             raise ValueError("l1_ratio must be in [0, 1]")
+        if not (0.0 < self.quantile < 1.0):
+            raise ValueError("quantile must be in [0, 1]")
```

Comment:
Could you use `check_scalar` here. No need to modify the other parameter

Reply:
Sure I will use `check_scalar` but I find there is a need to modify the parameters.

Reply:
> there is a need to modify the parameters.

What I mean is that do not validate all other parameters with `check_scalar`, just `quantile`. It will not be consistent but the meta-issue for `check_scalar` will come back to this class in a later PR.

Reply:
makes sense to me now.